openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIEndpointMapper.java

```diff
@@ -113,15 +114,25 @@ private Server addEnumValues(List<Server> servers) {
      */
     private Server extractServer(ListenerDeclarationNode ep, String serviceBasePath) {
         Optional<ParenthesizedArgList> list;
-        if (ep.initializer().kind().equals(SyntaxKind.EXPLICIT_NEW_EXPRESSION)) {
-           ExplicitNewExpressionNode bTypeExplicit = (ExplicitNewExpressionNode) ep.initializer();
+        if (ep.initializer().kind().equals(SyntaxKind.CHECK_EXPRESSION)) {
+            ExpressionNode expression = ((CheckExpressionNode) ep.initializer()).expression();
+            list = extractListenerNodeType(expression);
+        } else {
+            list = extractListenerNodeType(ep.initializer());
+        }
+        return generateServer(serviceBasePath, list);
+    }
+
+    private Optional<ParenthesizedArgList> extractListenerNodeType(Node expression2) {
+        Optional<ParenthesizedArgList> list;
+        if (expression2.kind().equals(SyntaxKind.EXPLICIT_NEW_EXPRESSION)) {
+            ExplicitNewExpressionNode bTypeExplicit = (ExplicitNewExpressionNode) expression2;
             list = Optional.ofNullable(bTypeExplicit.parenthesizedArgList());
         } else {
-            ImplicitNewExpressionNode  bTypeInit = (ImplicitNewExpressionNode) ep.initializer();
+            ImplicitNewExpressionNode bTypeInit = (ImplicitNewExpressionNode) expression2;
```

Comment:
Shouldn't we add a check to verify that the node kind is `IMPLICIT_NEW_EXPRESSION` before direct casting?